platforms/core-runtime/instrumentation-reporting/src/integTest/groovy/org/gradle/internal/instrumentation/reporting/PropertyUpgradeReportingIntegrationTest.groovy

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright 2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.instrumentation.reporting
+
+import groovy.test.NotYetImplemented
+import org.gradle.integtests.fixtures.AbstractIntegrationSpec
+import org.gradle.util.internal.ToBeImplemented
+
+class PropertyUpgradeReportingIntegrationTest extends AbstractIntegrationSpec {
+
+    def "usage of upgraded properties in buildSrc should be reported"() {
+        given:
+        executer.requireOwnGradleUserHomeDir("We cache report in global cache")
+        javaFile("buildSrc/src/main/java/test/MyPlugin.java", """
+            package test;
+
+            import org.gradle.api.Plugin;
+            import org.gradle.api.Project;
+            import org.gradle.api.tasks.compile.JavaCompile;
+
+            public abstract class MyPlugin implements Plugin<Project> {
+                @Override
+                public void apply(Project project) {
+                    project.getTasks().register("myJavaCompile", JavaCompile.class, task -> {
+                        task.getSource();
+                        task.setSource(project.files());
+                    });
+                }
+            }
+        """)
+        buildFile << """
+            apply plugin: test.MyPlugin
+        """
+
+        when:
+        run("--property-upgrade-report")
+
+        then:
+        postBuildOutputContains("org.gradle.api.tasks.compile.JavaCompile.getSource(): at test.MyPlugin(MyPlugin.java:12)")
+        postBuildOutputContains("org.gradle.api.tasks.compile.JavaCompile.setSource(): at test.MyPlugin(MyPlugin.java:13)")
```

Comment:
I actually chose `JavaCompile.source` property since I have a feeling that we will migrate it as one of the latest property. And thus we will have less problems on the demo branch.

Note that once we migrate all properties, we will anyway remove this report. So in Gradle 9 it probably doesn't make sense to have property upgrade report anyway, since all properties will be migrated.

Reply:
Hm... I wonder how we could keep the reporting logic, though. I mean, in Gradle 10 we might want to do something similar again...